core/services/keeper/registry_synchronizer_sync_test.go

```diff
@@ -0,0 +1,75 @@
+package keeper
+
+import (
+	"errors"
+	"fmt"
+	"math/big"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap/zapcore"
+
+	"github.com/smartcontractkit/chainlink/core/internal/testutils"
+	"github.com/smartcontractkit/chainlink/core/logger"
+	"github.com/smartcontractkit/chainlink/core/services/keystore/keys/ethkey"
+	"github.com/smartcontractkit/chainlink/core/utils"
+)
+
+// GetUpkeepFailure implements the UpkeepGetter interface with an induced error and nil
+// config response.
```

Comment:
You can probably implement this test inside `registry1_1_synchronizer_test` or `registry1_2_synchronizer_test`

Reply:
That was my first thought too, but both of those files declare the `keeper_test` package. This isolates all tests to the surface area of the 'keeper' package API. Because I needed to test a private member function, I needed to add the test to the `keeper` package directly.

My suggestion would be to move all the surface area testing into another file or set of files that clearly states the meaning and intent of this type of testing. Unit tests for any given file should be in `[filename]_test.go`, as you point out.

Reply:
We also have this doc on [testing guidelines](https://www.notion.so/chainlink/Core-Engine-Automated-Testing-Guidelines-4690e96fab40403483b74ca62704774a), see if it matches your expectations?